theme/templates/includes/my-resources-trows.html

```diff
@@ -0,0 +1,109 @@
+                    class="glyphicon glyphicon-star btn-inline-favorite isfavorite"></span>
+
+            <form class="hidden-form" data-id="form-favorite-{{ res.short_id }}"
+                    action="/hsapi/_internal/{{ res.short_id }}/label-resource-action/"
+                    method="POST">
+                {% csrf_token %}
+                <input type="hidden" name="action" value="DELETE">
+                <input type="hidden" name="label_type" value="FAVORITE">
+            </form>
+        {% else %}
+            <span data-form-id="form-favorite-{{ res.short_id }}"
+                    data-form-type="toggle-favorite"
+                    class="glyphicon glyphicon-star btn-inline-favorite"></span>
+
+            <form class="hidden-form" data-id="form-favorite-{{ res.short_id }}"
+                    action="/hsapi/_internal/{{ res.short_id }}/label-resource-action/"
+                    method="POST">
+                {% csrf_token %}
+                <input type="hidden" name="action" value="CREATE">
+                <input type="hidden" name="label_type" value="FAVORITE">
+            </form>
+        {% endif %}
+
+        <span class="glyphicon glyphicon-tag btn-inline-label" data-toggle="dropdown"
+                aria-expanded="false"></span>
+
+        <div class="dropdown-menu inline-dropdown" role="menu">
+            <div class="panel-body" role="form">
+                <ul data-resource-id="{{ res.short_id }}" class="list-group list-labels">
+                </ul>
+            </div>
+        </div>
+    </td>
+    {# Type #}
+    <td data-col="resource-type">
+        {% include "includes/res_type_col.html" with resource=res %}
+    </td>
+    {# Title #}
+    {% with metadata=res.metadata %}
+        <td>
+            <strong><a href="{{ res.get_absolute_url }}">{{ metadata.title|escape }}</a>
+            </strong>
+        </td>
+        {# First Author #}
+        <td>{{ res|resource_first_author }}</td>
+
+        {# Date Created #}
+        <td>{{ res.created|date:"M d, Y" }} at {{ res.created|time }}</td>
+        {# Last Modified #}
+        {% with updated=res.updated %}
+            <td>{{ updated|date:"M d, Y" }} at {{ updated|time }}</td>
+            <td>
+                {% for kw in metadata.subjects.all %}
+                    {% if forloop.counter0 > 0 %},{% endif %}{{ kw.value}}
+                {% endfor %}
+            </td>
+            <td>
+                {% for creator in metadata.creators.all %}
+                    {% if forloop.counter0 != 0 %}<span> · </span>{% endif %}
+                    {% if creator.relative_uri and creator.is_active %}
+                        <a href="{{ creator.relative_uri }}">{{ creator.name }}</a>
+                    {% else %}
+                        <span>{{ creator.name }}</span>
+                    {% endif %}
+                {% endfor %}
+            </td>
+            <td>{{ res|get_user_privilege:user }}</td>
```

Comment:
believe this is the only change in the template, using the tag to load privileges (this replaces the annotations that were previously done in the querysets

Reply:
I think this new filter `get_user_privilege:user` will generate a separate query for each resource. Using the annotation in queryset for user privilege avoids this additional query.